### DISCUSSION TITLE
Lesson 11 :  Type Errors

### DISCUSSION THREAD

**Discussion by TobiYusuf** [smartcontractkit/full-blockchain-solidity-course-py#1136] [Q&A] [answered]
Hello everyone, 

I have been coming across this error TypeError: Contract {nameOfContract} should be marked as abstract. So far in the course there hasn't been any mention of abstract contract yet, if I add abstract to the contract the error goes away.  What does it mean ?

Another typerror i get is - TypeError: Function has override specified but does not override anything. If I remove override from the function, the error goes away. Please can anyone explain this errors.

![Github Code](https://user-images.githubusercontent.com/59777469/155269721-8a4c92d6-ac1c-4414-a762-2ee0cde80178.PNG)

![Github code-terminal](https://user-images.githubusercontent.com/59777469/155269733-e6cfdf12-cc8e-4b51-9c37-f161a970ed37.PNG)

**Comment by cromewar** [comment, marked as answer]
Hello @TobiYusuf 
An abstract contract (object) is something that could no be used by itself and should be used on another contract, basically is creating a contract with functions without specifying what those functions do.
So, let's suppose you have an abstract contract (all interfaces are abstract) with 5 functions, and you import it from another contract you would have to declare what each of does functions does on your new contract, otherwise the functionality will be incomplete. If that happen, the compiler complains saying that unless you implement all the methods (function) the new contract should be marked as abstract as well.
Here is a more detailed explanation:
https://docs.soliditylang.org/en/v0.6.2/contracts.html

Also I also had the override problem, for me was i had a incorrect indentation on the code and I had to restart the editor.

I hope this info might help

**Reply by TobiYusuf** [reply]
Very helpful, thank you for the reply
